ta_services/secure_key_services/src/pkcs11_attributes.c

```diff
@@ -1485,3 +1485,50 @@ bool object_is_private(struct sks_attrs_head *head)
 
 	return false;
 }
+
+uint32_t generate_id(struct sks_attrs_head **attrs, uint32_t *set_id) {
+	uint32_t rv;
+	uint32_t id;
+
+	/* Check if SKS_CKA_ID is already present */
+	rv = get_attribute(*attrs, SKS_CKA_ID, NULL, NULL);
+	if (rv != SKS_NOT_FOUND)
+		return SKS_OK;
+	
+	/* Generate 32 bit ID */
+	if (!set_id)
+		TEE_GenerateRandom(&id, sizeof(id));
```

Comment:
no issue with collision? should it insure ID unique?

Reply:
I added this, so OpenSSH would operate on the keys I generated (since it would refuse if there was no CKA_ID). It would be nice to check for collisions, but at the time I was writing this I couldn't come up with any easy way to do so and later forgot about it. The standard also doesn't seem to require us to ensure this:
> The CKA_ID attribute is intended as a means of distinguishing multiple public-key/private-key pairs held by the same subject (whether stored in the same token or not). (Since the keys are distinguished by subject name as well as identifier, it is possible that keys for different subjects may have the same CKA_ID value without introducing any ambiguity.)
> It is intended in the interests of interoperability that the subject name and key identifier for a certificate will be the same as those for the corresponding public and private keys (though it is not required that all be stored in the same token). However, Cryptoki does not enforce this association, or even the uniqueness of the key identifier for a given subject; in particular, an application may leave the key identifier empty.

In any case, I will look into how I can iterate through all objects present in the SKS and implement this check.